### PR TITLE
GG-34696 Wrong calculation of replies count for queries with partitio…

### PR DESCRIPTION
**Pull request by korlov42**
…n pruning and enabled parallelism

### CODE REVIEW

**Review by tledkov — APPROVED** [review]
OK with me

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java [hunk 1]
```diff
@@ -610,6 +613,51 @@ else if (QueryUtils.wasCancelled(err))
...
+                assert prev == null;
+            }
+
+            return res;
+        }
+
+        final int segments = mapper.findFirstPartitioned(qry.cacheIds()).config().getQueryParallelism();
+
+        for (ClusterNode node : nodes) {
+            Map<ClusterNode, IntArray> partsMapping = mapping.queryPartitionsMap();
```
Comment (zstan) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java:641]: escape analysis of course will help here, but you can explicitly move **partsMapping** within the loop scope.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public void testWhereCounteryPartitionQuery() throws Exception {
      * Check case with 2 partitions. Multiple partitions should not be affected.
      */
     @Test
-    public void testWhereCountMultiPartitionsQuery() throws Exception {
+    public void testWhereCountMultiPartitionsQuery() {
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlSinglePartitionMultiParallelismTest.java:108]: we need parametrized tests here

**Review by zstan — APPROVED** [review]
I fill some comments, overall looks good.

**Pull request merged**